### DISCUSSION TITLE
How can we get rid of the "setsockopt socket 3 IPV6_TCLASS 16: Operation not permitted" SSH warnings in WSL1?

### DISCUSSION THREAD

**Discussion by catharsis71** [microsoft/WSL#8718] [Q&A]
What can be done about these warnings when using SSH or anything that relies on SSH?

```
setsockopt socket 3 IPV6_TCLASS 16: Operation not permitted:
setsockopt socket 3 IPV6_TCLASS 16: Operation not permitted:
```

or like this

```
setsockopt socket 3 IPV6_TCLASS 16: Operation not permitted:
setsockopt socket 3 IPV6_TCLASS 8: Operation not permitted:
```

Based on some Googling it appears this may be "fixed" in WSL2 but WLS2 is not usable yet due to lack of IPV6 support.

The suggested workaround ```-oAddressFamily=inet``` isn't usable because it prevents connecting to IPV6-only hosts

and using ```-oAddressFamily=inet6``` instead does not suppress the warnings

I upgraded from Ubuntu LTS 20.04 to 22.04 today and now I'm seeing even more of the warnings than before

**Comment by NotTheDr01ds** [comment]
I don't have an answer for you, but just a warning that there are other issues on 22.04 on WSL1.  Specifically, `gzip` is [broken without manual patching](https://github.com/microsoft/WSL/issues/8219).  Node (and others) also have the issue.
